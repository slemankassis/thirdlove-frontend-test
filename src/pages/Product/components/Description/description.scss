@import "./../../../../styles/variables.scss";

.product-description {
  width: 320px;
  height: 389px;
  font-family: Montserrat;
  font-size: 13px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.77;
  letter-spacing: normal;
  color: #444444;

  @media (min-width: $breakpoint-tablet) {
    display: grid;
  }

  @media (min-width: $breakpoint-desktop) {
    display: grid;
  }
}
