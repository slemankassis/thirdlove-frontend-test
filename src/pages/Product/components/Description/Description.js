import React from 'react';
import { sanitizeHtml } from '../../../../utils';
import './description.scss';

class Description extends React.Component {
  render() {
    return (
      <div className={this.props.className}>
        {sanitizeHtml(this.props.contentHtml, ['meta', 'p', 'ul', 'li', 'span'])}
      </div>
    );
  }
}

export default Description;
