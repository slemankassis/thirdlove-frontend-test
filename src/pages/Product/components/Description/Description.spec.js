import React from 'react';
import Description from './Description';

describe('Description snaps', () => {
  it('renders snapshot', () => {
    const props = {
      contentHtml: mockData.product.body_html,
    };

    const component = shallow(<Description {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
