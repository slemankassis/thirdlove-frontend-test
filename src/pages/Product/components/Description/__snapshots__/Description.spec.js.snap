// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description snaps renders snapshot 1`] = `
"<div className=''>
  <Label className='product-description__label' text='DETAILS' />
  <div dangerouslySetInnerHTML={{...}} />
</div>"
`;
